### PR TITLE
Align status codes for errors in Application Gateway

### PR DESCRIPTION
**Pull request by mvshao**
**Description**
REST service should return 5xx for errors that occurred on the server side and 4xx for errors on the client side. Error code should be as informative as possible. 

Changes proposed in this pull request:

- `404 Not Found` status code when the application specified in the path doesn't exist
- `400 Bad Request` status code when application, service or entry for the Compass use case is not specified in the path
- `504 Gateway Timeout` when call to target API timeouts Application Gateway
- deletion of the `WriteTimeout` from the Internal Server and Internal Compass Server due to inconsistencies in timeouts. It was done to handle correctly timeout code rewriting.

**Related issue(s)**
#14842

### CODE REVIEW

**Review by VOID404 — CHANGES_REQUESTED** [review]

File: components/central-application-gateway/internal/proxy/reverseproxy.go [hunk 1]
```diff
@@ -54,7 +55,10 @@ func newProxy(targetURL string, requestParameters *authorization.RequestParamete
 
 		log.Infof("Modified request url : '%s', schema : '%s', path : '%s'", req.URL.String(), req.URL.Scheme, req.URL.Path)
 	}
-	return &httputil.ReverseProxy{Director: director, Transport: transport}, nil
+	errorHandler := func(rw http.ResponseWriter, req *http.Request, err error) {
+		codeRewriter(rw, err)
+	}
```
Comment (VOID404) [inline components/central-application-gateway/internal/proxy/reverseproxy.go:60]: Aren't we missing logging now?

Comment (mvshao) [reply components/central-application-gateway/internal/proxy/reverseproxy.go:60]: I will add necessary logging 👍🏻

File: components/central-application-gateway/internal/proxy/reverseproxy.go [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 
-	log "github.com/sirupsen/logrus"
-
 	"github.com/kyma-project/kyma/components/central-application-gateway/internal/csrf"
 	"github.com/kyma-project/kyma/components/central-application-gateway/pkg/apperrors"
 	"github.com/kyma-project/kyma/components/central-application-gateway/pkg/authorization"
 	"github.com/kyma-project/kyma/components/central-application-gateway/pkg/authorization/clientcert"
 	"github.com/kyma-project/kyma/components/central-application-gateway/pkg/httpconsts"
 	"github.com/kyma-project/kyma/components/central-application-gateway/pkg/httptools"
+	log "github.com/sirupsen/logrus"
```
Comment (VOID404) [inline components/central-application-gateway/internal/proxy/reverseproxy.go:17]: Unsorting imports is a weird change to include :P

File: components/central-application-gateway/internal/proxy/reverseproxy.go (hunk 1 above)
Comment (VOID404) [reply components/central-application-gateway/internal/proxy/reverseproxy.go:60]: And what about logging other errors?
Also we *could* take into account `ErrorLog` field of the `ReverseProxy`, 
but I don't feel like it's really necessary if we're not configuring that field.

**Review by VOID404 — CHANGES_REQUESTED** [review]

File: components/central-application-gateway/internal/proxy/reverseproxy.go (hunk 1 above)
Comment (mvshao) [reply components/central-application-gateway/internal/proxy/reverseproxy.go:60]: Other errors are handled by a default error handler in the reverse proxy package.

**Review by VOID404 — APPROVED** [review]

**Pull request merged**